core/tools/coverage-report

```diff
@@ -1,25 +1,72 @@
 #!/bin/sh
 
-COVERAGE_THRESHOLD=${COVERAGE_THRESHOLD:-0}
+# Configurable variables
+readonly COVERAGE_THRESHOLD=${COVERAGE_THRESHOLD:-0}
+readonly WHOLE_SRC_COVERAGE=${WHOLE_SRC_COVERAGE:-1}
+readonly MOVE_FROM_SRC=${MOVE_FROM_SRC:-1}
 
-coverage run --source=./src /dev/null 2>/dev/null
-mv .coverage .coverage.empty
-coverage combine .coverage.*
+if [ ${COVERAGE_THRESHOLD} -ne 0 ]; then
+  echo "COVERAGE_THRESHOLD set to ${COVERAGE_THRESHOLD}%"
+fi
+
+# Optionally move coverage files from src
+# (tests usually save it there).
+if [ ${MOVE_FROM_SRC} -eq 1 ]; then
+  if ls src/.coverage* 2>/dev/null; then
+    echo "Moving coverage files from src"
+    mv src/.coverage* .
+  fi
+fi
+
+# When there are some .coverage.* files, it is a sign
+# we need to combine them into one .coverage file.
+if ls .coverage.* 2>/dev/null; then
+  # Need to create an empty coverage file containing
+  # all existing code files, to get coverage over whole src.
+  if [ ${WHOLE_SRC_COVERAGE} -eq 1 ]; then
```

Comment:
in what situation would we want to skip this step?

Reply:
istm in such scenario an output that tells you the list of items that you are _not_ currently covering is overall more correct?